genesyscloud/journey_action_map/genesyscloud_journey_action_map_init_test.go

```diff
@@ -0,0 +1,82 @@
+package journey_action_map
+
+import (
+	"log"
+	"sync"
+	gcloud "terraform-provider-genesyscloud/genesyscloud"
+	"terraform-provider-genesyscloud/genesyscloud/architect_flow"
+	"terraform-provider-genesyscloud/genesyscloud/architect_schedulegroups"
+	"terraform-provider-genesyscloud/genesyscloud/architect_schedules"
+	authDivision "terraform-provider-genesyscloud/genesyscloud/auth_division"
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v146/platformclientv2"
+)
+
+var (
+	sdkConfig *platformclientv2.Configuration
+	err       error
+)
+
+/*
+   The genesyscloud_journey_action_map_init_test.go file is used to initialize the data sources and resources
+   used in testing the journey_action_map resource.
+*/
+
+// providerDataSources holds a map of all registered datasources
+var providerDataSources map[string]*schema.Resource
+
+// providerResources holds a map of all registered resources
+var providerResources map[string]*schema.Resource
+
+type registerTestInstance struct {
+	resourceMapMutex   sync.RWMutex
+	datasourceMapMutex sync.RWMutex
+}
+
+// registerTestResources registers all resources used in the tests
+func (r *registerTestInstance) registerTestResources() {
+	r.resourceMapMutex.Lock()
+	defer r.resourceMapMutex.Unlock()
+
+	providerResources[resourceName] = ResourceJourneyActionMap()
+	providerResources["genesyscloud_auth_division"] = authDivision.ResourceAuthDivision()
```

Comment:
You can now refer to the provider resource types like so: instead of `"genesyscloud_auth_division"` you can use `authDivision.ResourceType` and the same for all the other non-`gcloud` resources.